src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -206,10 +206,20 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!-- Add conditional compilation symbols for target platform backwards compatibility. -->
+  <Target Name="GeneratePlatformCompatibleDefineConstants"
+          Condition=" '$(DisableImplicitFrameworkDefines)' != 'true' and '$(TargetPlatformIdentifier)' != '' and '$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(TargetFrameworkVersion), 5.0)) " >
+    <ItemGroup>
+      <_SupportedPlatformCompatibleVersions Include="@(SupportedTargetPlatform)" Condition=" $(TargetPlatformMinVersion) != '' and %(Identity) != '' and $([MSBuild]::VersionGreaterThanOrEquals(%(Identity), $(TargetPlatformMinVersion))) and $([MSBuild]::VersionLessThan(%(Identity), $(TargetPlatformVersion))) " />
```

Comment:
I'm not following what this logic is supposed to be doing with `TargetPlatformMinVersion`.  However, I think it should not be based on that property at all.  It should simply define constants for all supported versions less than or equal to the `TargetPlatformVersion`, pretty much the same way that it works for the `TargetFrameworkVersion`.

IE, I think that [this section of the spec](https://github.com/dotnet/designs/blob/master/accepted/2020/net5/net5.md#preprocessor-symbols) is only talking about `TargetPlatformVersion`, not the minimum version. @terrajobst @mhutch

Reply:
Sounds good, I've pushed a fix.